### DISCUSSION TITLE
Pivot and accessible from Logica

### DISCUSSION THREAD

**Discussion by jedi1992** [EvgSkv/logica#128] [Q&A]
Hi ,
I was trying this example https://medium.com/google-cloud/building-sql-pipelines-in-bigquery-with-dataform-part-1-9e96f14ec664

a) Have to use the pivot operator to have the final predicate in the form of 
Date, temp_min, temp_max, rain   for a  weather station (already filtered in) . After the pivot I have a 4 column table  but since I have used pivot operator I cannot treat it like a predicate and do further processing in Logica. Have to default to Python/Panda.  Would love to continue being in the Logica domain itself. 
Can I import back the pivoted table as if were a new predicate in logica and then do processing on top of it using Logica syntax ?

b) Also in the same example they do a group by Date and do 
  MAX(prcp) AS rain_mm,
  MIN(tmin) AS tmin_celsius,
  MAX(tmax) AS tmax_celsius

In logica the min,max syntax is not as easy as say other aggregators like  count, set, average, list etc. Not sure if I am missing a trick here.  Still struggling with max, group by date syntax. Any pointers would be welcome.

Thanks

**Comment by EvgSkv** [comment]
Hello,

a) You could write the pivot table back into BQ dataset and query as any other table. Would this work for you?
You can also call SQL from Logica, but it could be cumbersome. Example:

```
@With(T);
T(..r) :- `((
  SELECT 1 AS X UNION ALL
  SELECT 2 AS X
))`(..r);

Q(x, y) :- T(x:), T(x: y);
```

b) There are `Min` and `Max` aggregating functions in Logica. Why do you think it's as easy to use them as other aggregations?

Please let me know if you have further questions!
